core/src/main/java/com/scalar/db/util/ActiveExpiringMap.java

```diff
@@ -81,8 +81,12 @@ public boolean contains(K key) {
     return map.containsKey(key);
   }
 
-  public void remove(K key) {
-    map.remove(key);
+  public V remove(K key) {
+    ValueHolder<V> prev = map.remove(key);
+    if (prev == null) {
+      return null;
+    }
+    return prev.get();
   }
 
   public void updateExpirationTime(K key) {
```

Comment:
This method does not appear to be used anymore, should we still keep it?

Reply:
Thank you for pointing it out!

I think we can keep it because it's a useful for some use cases. `ActiveExpiringMap` is a commonly used utility class and it might be used in another part in the future.